lib/init.js

```diff
@@ -0,0 +1,74 @@
+const EE = require('events').EventEmitter
+const { isMainThread } = require('./worker_threads')
+
+const initWorkers = require('./manager')
+const validateOpts = require('./validate')
+const noop = require('./noop')
+const runTracker = require('./runTracker')
+const track = require('./progressTracker')
+
+function init (opts, cb) {
+  return _init(opts, cb)
+}
+
+function _init (opts, cb, tracker) {
+  const cbPassedIn = (typeof cb === 'function')
+  cb = cb || noop
+
+  tracker = tracker || new EE()
+
+  if (!cbPassedIn && !opts.forever) {
+    const promise = new Promise((resolve, reject) => {
+      _init(opts, (err, results) => {
+        if (err) return reject(err)
+        resolve(results)
+      }, tracker)
+    })
+    tracker.then = promise.then.bind(promise)
+    tracker.catch = promise.catch.bind(promise)
+    return tracker
+  }
+
+  opts = validateOpts(opts, cbPassedIn)
+
+  function _cb (err, result) {
+    if (err) {
+      return cbPassedIn ? cb(err) : setImmediate(() => tracker.emit('error', err))
+    }
+
+    setImmediate(() => tracker.emit('done', result))
+    cb(null, result)
+
+    if (!err && opts.json) {
+      console.log(JSON.stringify(result))
```

Comment:
Yes it was there before.
I agree, but since it's specific to `json` option, I thought it might come in handy for someone who intentionally wants json to be printed.
But let me know what would you'd do instead.